components/style/selector_matching.rs

```diff
@@ -747,12 +756,20 @@ impl SelectorMap {
                                 parent_bf: Option<&BloomFilter>,
                                 rules: &[Rule],
                                 matching_rules: &mut V,
-                                relations: &mut StyleRelations)
+                                relations: &mut StyleRelations,
+                                importance: Importance)
         where E: Element<Impl=TheSelectorImpl>,
               V: VecLike<DeclarationBlock>
     {
         for rule in rules.iter() {
-            if matches_complex_selector(&*rule.selector,
+            let block = &rule.declarations.mixed_declarations;
+            let any_declaration_for_importance = if importance.important() {
+                block.important_count > 0
+            } else {
+                block.declarations.len() > block.important_count as usize
```

Comment:
I think `block.declarations.len() != block.important_count` would be clearer (or maybe adding a helper method to get the normal count?)